### PR TITLE
add model run parameter tables

### PR DESCRIPTION
**Pull request by littledice**

### CODE REVIEW

File: migrations/sql/V2017.14.11.1716__AddModelRunParameters.sql [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+ALTER TABLE model_run ADD FOREIGN KEY (model_run_parameter_set) REFERENCES model_run_parameter_set.id;
+
+CREATE TABLE model_run_parameter(
+model_run INTEGER NOT NULL,
+key TEXT NOT NULL,
+value TEXT NOT NULL
+)
+
+ALTER TABLE model_run_parameter ADD FOREIGN KEY (model_run) REFERENCES model_run.id;
```
Comment (MartinEden) [inline migrations/sql/V2017.14.11.1716__AddModelRunParameters.sql]: Should be `REFERENCES model_run.internal_id`, and it should be a serial to match.

File: migrations/sql/V2017.14.11.1716__AddModelRunParameters.sql [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+);
+
+ALTER TABLE model_run_parameter_set ADD FOREIGN KEY (responsibility_set) REFERENCES responsibility_set.id;
+ALTER TABLE model_run_parameter_set ADD FOREIGN KEY (model_version) REFERENCES model_version.id;
+ALTER TABLE model_run_parameter_set ADD FOREIGN KEY (upload_info) REFERENCES upload_info.id;
+
+CREATE TABLE model_run(
+intenal_id SERIAL,
+run_id TEXT NOT NULL,
+model_run_parameter_set INTEGER NOT NULL
```
Comment (MartinEden) [inline migrations/sql/V2017.14.11.1716__AddModelRunParameters.sql]: Should be a serial, I think

Comment (littledice) [reply migrations/sql/V2017.14.11.1716__AddModelRunParameters.sql]: this one's a foreign key to another table, we don't need/want it to autoincrement?

Comment (MartinEden) [reply migrations/sql/V2017.14.11.1716__AddModelRunParameters.sql]: You're right. Sorry; I thought `serial` was more than just an auto-incrementing integer.

File: migrations/sql/V2017.14.11.1716__AddModelRunParameters.sql [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+PRIMARY KEY (internal_id)
+);
+
+ALTER TABLE model_run ADD FOREIGN KEY (model_run_parameter_set) REFERENCES model_run_parameter_set (id);
+
+CREATE TABLE model_run_parameter(
+model_run INTEGER NOT NULL,
+key TEXT NOT NULL,
+value TEXT NOT NULL,
+UNIQUE (key, model_run)
```
Comment (MartinEden) [inline migrations/sql/V2017.14.11.1716__AddModelRunParameters.sql]: Maybe this should be a clustered primary key, and not just a unique constraint? Either that are we should maybe just add an artificial integer primary key. It seems weird to have no primary key at all.

Comment (littledice) [reply migrations/sql/V2017.14.11.1716__AddModelRunParameters.sql]: Yeah I was wondering about that, lets go with integer key then

**Comment by MartinEden** [comment]
Happy for you to merge this once it passes in TeamCity.

**Pull request merged**